ext/hash/hash.c

```diff
@@ -632,12 +637,13 @@ PHP_FUNCTION(hash_init)
 	hash = php_hashcontext_from_object(Z_OBJ_P(return_value));
 
 	context = php_hash_alloc_context(ops);
-	ops->hash_init(context);
+	ops->hash_init(context, args);
 
 	hash->ops = ops;
 	hash->context = context;
 	hash->options = options;
 	hash->key = NULL;
+	hash->args = args && zend_hash_num_elements(args) > 0 ? zend_array_dup(args) : NULL;
```

Comment:
Still doing a zend_array_dup here, incrementing refcount would be sufficient.

Reply:
Indeed. Fixed, thanks.